GUIDE_LOCALIZATION.md

```diff
@@ -1,68 +1,97 @@
 # Advanced Guide: Localization
 
-This guide assumes you have already completed the beginner's guide and are familiar with the basics of setting up and running a Podium podlet server. In this advanced guide, we will cover how to add localization support to your Podium podlet application using the built-in localization features.
+This guide assumes you have already completed the beginner's guide and are familiar with the basics of setting up and running a Podium podlet server. In this advanced guide, we will cover how to add localization support to your Podium podlet application using the built-in localization features. Localization support is built on top of [lingui](https://lingui.dev/).
 
-## Step 1: Create the locale directory
+## Step 1: Set up your localization config
 
-Create a new directory named locale in the root of your project. This is where you will store your localization JSON files.
+Create a new file called `.linguirc` that will hold your localization config.
 
-```sh
-mkdir locale
+```jsonc
+{
+  "locales": ["en-US", "no-NB"], // what locales/languages to support
+  "sourceLocale": "en-US", // the source locale (language used in the code)
+  // catalogs tells us where to look for, and where to output message files
+  // the podlet server will always look for locale files on the path used below
+  "catalogs": [
+    {
+      "include": ["<rootDir>/content.js"],
+      "path": "<rootDir>/locales/{locale}/messages"
+    }
+  ],
+  // use named exports in compiled files (must be set to ES)
+  "compileNamespace": "es"
+}
```

Comment:
Is there a programmatic API? I'm wondering if there is a way we can roll all of this into the podlet server so that (in at least most cases), users don't need to define a config file at all and things just work. Eg. generate po files etc when the server is started up. If Lingui has a non compile mode we could also look at using that when the server is in dev mode and only compile when you run a build etc.

Reply:
> 

It's not exported for first hand usage as programmatic API, but we could potentially create a nicer wrapper around the CLI indeed. I can explore this more today.